comment_limit.module

```diff
@@ -192,36 +196,44 @@ function comment_limit_entity_view(array &$build, EntityInterface $entity, Entit
 
   foreach ($fields as $delta => $field) {
     if ($field->getType() == 'comment') {
+
       $comment_limit = Drupal::service('comment_limit.service');
-      $entity_id = $entity->id();
-      $entity_type = $entity->getEntityTypeId();
-      $field_name = $field->getName();
-      $field_id = $field->id();
-      $field_label = $field->getLabel();
-      $field_limit = $comment_limit->getFieldLimit($field_id);
-      $user_limit = $comment_limit->getUserLimit($field_id);
-      $field_current = $comment_limit->getCurrentCommentsOnField($entity_id, $entity_type, $field_name);
-      $user_current = $comment_limit->getCurrentCommentCountForUser($entity_id, $entity_type, $field_name);
-      if ($display->getComponent('field_limit')) {
-        $build[$delta . 'field_limit'] = [
+      $field_limit = $comment_limit->getFieldLimit($field->id());
+      $user_limit = $comment_limit->getUserLimit($field->id());
+      $field_current_count = $comment_limit->getCurrentCommentsOnField(
+        $entity->id(),
+        $entity->getEntityTypeId(),
+        $field->getName()
+      );
+      $user_current_count = $comment_limit->getCurrentCommentCountForUser(
+        $entity->id(),
+        $entity->getEntityTypeId(),
+        $field->getName()
+      );
+
+      if ($component = $display->getComponent('field_limit')) {
+        $build['comment_limit_field_limit'] = [
           '#type' => 'item',
-          '#markup' => '<b>' . $this->t('The comment field "%field" is generally limited to a maximum of %maximum comments, %count have already been posted.', [
-            '%field' => $field_label,
+          '#weight' => $component['weight'],
+          '#markup' => '<em>' . t('The comment field "%field" is generally limited to a maximum of %maximum comments, %count have already been posted.', [
+            '%field' => $field->getLabel(),
             '%maximum' => $field_limit,
-            '%count' => $field_current,
-          ]) . '</b>',
+            '%count' => $field_current_count,
+          ]) . '</em>',
         ];
       }
-      if ($display->getComponent('user_limit')) {
-        $build[$delta . 'user_limit'] = [
+      if ($component = $display->getComponent('user_limit')) {
+        $build['comment_limit_user_limit'] = [
```

Comment:
This won't work on multiple fields, because it will just overwrite $build['comment_limit_user_limit'] instead of adding a new one.